### ISSUE TITLE
Allow to hide the bottom 2D UI bar

### ISSUE THREAD

**Issue by takahirox** [mozilla/hubs#5188]
**Is your feature request related to a problem? Please describe.**

The bottom 2D UI bar may be too tall on some platforms, especially on mobile devices with landscape mode.

If users use the virtual joysticks the tall UI bar can be more problematic. The UI bar makes the joysticks too high up on the screen.

![image](https://user-images.githubusercontent.com/7637832/155801683-cf076a7a-cc8a-40d4-8a0f-7f51ecfa30d0.png)

**Describe the solution you'd like**

Allow to hide the bottom 2D UI bar. Maybe click something that slides it down and off the screen, with a single small button that will bring it back.

**Comment by takahirox** [comment]
Related or duplicated with #3468

**Comment by takahirox** [comment]
An important thing would be Voice (mic) and Share should always visible for user privacy. They should always be able to be aware of mic on and screen/camera share on.

So, an easy solution would be transparent tool bar. It would be opacity if it's focused.

![image](https://user-images.githubusercontent.com/7637832/174418083-b9a0fe4d-3a84-4226-a6f6-e834da6fa1e3.png)

**Comment by vincentfretin** [comment]
Personally I'm not a fan of semi-transparent block that become solid on hover, and you can't really hover on touch devices. The low opacity value you used in your image is not accessible for people having bad eyes.
Using a semi-transparent like this doesn't resolve the fact you have less space in the scene part to move the avatar on mobiles.
I would prefer an arrow on left or right to minimize it and keep a smaller voice and share button maybe.
Or a button to make all the icons twice smaller and no text so reducing the bottom bar height by 2?

**Comment by vincentfretin** [comment]
Or avoid a button to minimize entirely, and use an alternate bottom bar for low resolutions, so mobiles and tablets.
Some folks did an alternate UI with fully transparent background, smaller icons with no text on their Hubs Cloud, there is some screenshots on discord dev-general.

**Comment by Exairnous** [comment]
I agree with what @vincentfretin has said, but I'd also like to mention that it can be hidden already on desktop via the backtick key and so I don't think the UI needs any modification on desktop, except maybe adding an option to the preferences to make this key configurable as not everyone has access to it.

**Comment by takahirox** [comment]
> it can be hidden already on desktop via the backtick key 

Oh, I didn't know that...

**Comment by vincentfretin** [comment]
> it can be hidden already on desktop via the backtick key and so I don't think the UI needs any modification on desktop, except maybe adding an option to the preferences to make this key configurable as not everyone has access to it.

On French keyboard, I need to use AltGr+7 to access the backtick. :D
The position of the backtick on qwerty keyboard is the first key on the top left. On French keyboard it's the ² character.
But I think the issue can be resolved for French keyboard to use the same keyboard key if you use keyup/keydown `event.code` instead of `event.keyCode` in the code. I did a similar fix for wasd vs zqsd on French keyboard some times ago.

**Comment by vincentfretin** [comment]
For zqsd, it was this PR for reference https://github.com/mozilla/hubs/pull/3617
Where is the backtick key handled in the code?

**Comment by matthewbcool** [comment]
`userinput.el.emit("action_toggle_ui")`

**Comment by Exairnous** [comment]
@takahirox Yeah, it's kinda a hidden feature, which is one reason I suggested adding a preference for it, to boost discoverability.

@vincentfretin Yup, mapping to the position, rather than the key, is one way to solve it, but it has a couple drawbacks that Dom mentioned.  If everything else is being handled via position currently, then it should probably follow suit (it may be being mapped by position already?), but it still would be nice if it were more discoverable.

**Comment by vincentfretin** [comment]
All the hubs code is using the key (character) for all keyboard interactions, with the exception of wasdqe keys, it's the position on the keyboard. Luckily it's the same set of keys on both azerty and qwerty. See
https://github.com/mozilla/hubs/blob/25e28a2a7f68b1627c92d07b362272949492a9bd/src/systems/userinput/devices/keyboard.js#L4-L12
and
https://github.com/mozilla/hubs/blob/25e28a2a7f68b1627c92d07b362272949492a9bd/src/systems/userinput/devices/keyboard.js#L60-L62

For backtick key to work on azerty, it's probably a matter of adding the correct key code in the list.

**Comment by Exairnous** [comment]
@vincentfretin Ah okay.  Thank you.  So it shouldn't be too hard to fix it that way then.  Unfortunately, it's still not really a solution for mobile, but perhaps something similar can be done, maybe a little arrow button on the white bar and then a double tap at the bottom edge of the screen?

**Comment by takahirox** [comment]
Another idea, full transparent (or no) tool bar background.

![image](https://user-images.githubusercontent.com/7637832/174898995-4f650a09-c9c0-4ef8-b14b-e11ad6105d13.png)

**Comment by Exairnous** [comment]
Interesting, that's actually something I suggested when I first gave feedback on the new UI design last year, making the bar transparent.  I've gotten used to the bar and just hide it when I want to, but it looks like it'll provide a lot better visibility on mobile.  I wonder if it would be possible to make the bar color (including alpha) a user preference, because I'm thinking that having it transparent could make it blend into the background of some scenes (especially that microphone button), and so reduce its accessibility.  Anyway, interesting ideas here.

**Comment by takahirox** [comment]
Currently I don't think of the preference. I may ask owners/admins to do that in their custom clients if they want.
